pages/_app.tsx

```diff
@@ -46,15 +48,12 @@ export default class MyApp extends App {
 
   render() {
     const {Component, pageProps} = this.props;
-    const {access, redirect, isPublic} = this.state;
+    const {access, redirect, isPublic, isAccessFetched} = this.state;
 
-    // if access and user values have a value now
-    if (access !== null && redirect !== null) {
-      // if user has access to page but there is a redirect
+    if (isAccessFetched) {
```

Comment:
Got it - so this is how you compensated for the change in the other two variables from `null` to `undefined` or `false` - this is fine.

Reply:
@lloan,

I'll try to answer all of your questions. In most Typescript work, _undefined_ is preferred to _null_. It's not a requirement, but it is the suggestion of the [Typescript Style Guide](https://github.com/Microsoft/TypeScript/wiki/Coding-guidelines#null-and-undefined)

The reason I did not want to use false for a redirect is more of a logical preference. I consider a redirect (the property of Redirect) to be a string if it is defined. I could have also declared the type of the redirect property to be string | boolean (or even string | boolean | undefined), but there is never a case where the redirect property would be true. Therefore, I felt undefined better captures when there isn't a redirect. 

Anyhow, in general I would try to avoid going into files like pages.tsx (with the redirects), but it was early enough on that I thought it would be ok. 

Sorry, no TLDR here. :)